File: goapi/internal/domain/nutrient.go
package domain

import "time"

type Nutrient struct {
    ID        uint      `json:"id" gorm:"primaryKey"`
    Name      string    `json:"name"`
    ParentID  *uint     `json:"parent_id"`
    CreatedAt time.Time `json:"created_at"`
    UpdatedAt time.Time `json:"updated_at"`
}

File: graphqlapi/src/backend/domain/entities/Nutrient.ts
// backend/domain/entities/Nutrient.ts

export interface Nutrient {
  id: number;
  name: string;
  code: string;
  parentId: string | null;
  createdAt: Date;
  updatedAt: Date;
}

export interface CreateNutrientRepositoryInput {
  name: string;
  code: string;
  parentId?: string | null;
}

export interface UpdateNutrientRepositoryInput {
  name?: string;
  code?: string;
  parentId?: string | null;
}
